packages/11ty/_plugins/transforms/outputs/epub/transform.js

```diff
@@ -143,12 +143,27 @@ module.exports = function(eleventyConfig, collections, content) {
     }
     if (!isPageLink(href)) return
 
+    const relativeURL = (href) => {
+      const hashRegexp = /(.*)(\#.*$)/
+      const match = href.match(hashRegexp)
+      const hash = match ? match[2] : ''
+      const pathname = match ? match[1] : href
+
+      return {
+        hash,
+        href,
+        pathname
+      }
+    }
+
+    const { hash, pathname } = relativeURL(href)
+
```

Comment:
We could use a `URL` instance here with `try` `catch` `finally` blocks...
```suggestion
    function relativeUrl (path) {
      const base = eleventyConfig.baseURL || 'http://localhost'
      let url;
      try {
        url = new URL(path)
      } catch (TypeError) {
        url = new URL(path, base)
      } finally {
        return url
      }
    }
    const { hash, pathname } = relativeUrl(href)

```